src/frontend/src/binder/insert.rs

```diff
@@ -135,6 +143,24 @@ impl Binder {
             ))));
         }
 
+        let (returning_list, aliases) = self.bind_select_list(returning_items)?;
+        if returning_list
+            .iter()
+            .any(|expr| expr.has_agg_call() || expr.has_window_function())
+        {
+            return Err(RwError::from(ErrorCode::BindError(
+                "INSERT should not have agg/window".to_string(),
+            )));
+        }
+
+        let fields = returning_list
+            .iter()
+            .zip_eq(aliases.iter())
+            .map(|(s, a)| {
+                let name = a.clone().unwrap_or_else(|| UNNAMED_COLUMN.to_string());
+                Ok(Field::with_name(s.return_type(), name))
+            })
+            .collect::<Result<Vec<Field>>>()?;
```

Comment:
How about introducing a light wrapper for this snippet, returning the `returning_list` and `fields`?